fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/GroupResourcePack.java

```diff
@@ -56,81 +52,58 @@ public GroupResourcePack(ResourceType type, List<ModResourcePack> packs) {
 	}
 
 	@Override
-	public InputStream open(ResourceType type, Identifier id) throws IOException {
+	public InputSupplier<InputStream> open(ResourceType type, Identifier id) {
 		List<ModResourcePack> packs = this.namespacedPacks.get(id.getNamespace());
 
 		if (packs != null) {
 			for (int i = packs.size() - 1; i >= 0; i--) {
 				ResourcePack pack = packs.get(i);
+				InputSupplier<InputStream> supplier = pack.open(type, id);
 
-				if (pack.contains(type, id)) {
-					return pack.open(type, id);
+				if (supplier != null) {
+					return supplier;
 				}
 			}
 		}
 
-		throw new ResourceNotFoundException(null,
-				String.format("%s/%s/%s", type.getDirectory(), id.getNamespace(), id.getPath()));
+		return null;
 	}
 
 	@Override
-	public Collection<Identifier> findResources(ResourceType type, String namespace, String prefix, Predicate<Identifier> predicate) {
+	public void findResources(ResourceType type, String namespace, String prefix, ResultConsumer arg) {
 		List<ModResourcePack> packs = this.namespacedPacks.get(namespace);
 
-		if (packs == null) {
-			return Collections.emptyList();
-		}
-
-		Set<Identifier> resources = new HashSet<>();
-
 		for (int i = packs.size() - 1; i >= 0; i--) {
 			ResourcePack pack = packs.get(i);
-			Collection<Identifier> modResources = pack.findResources(type, namespace, prefix, predicate);
 
-			resources.addAll(modResources);
+			pack.findResources(type, namespace, prefix, arg);
```

Comment:
Not sure if it’s an issue, but question: is it guaranteed that mod packs within a group won’t have intersecting resources? Might need to filter those duplicates before invoking the consumer 🙂

Reply:
As Gegy said, this absolutely needs filtering to de-dupe as there's no guarantee that mod resource packs won't have intersecting resources in the same group!

Reply:
....
Damn it Mojang!

Reply:
Hmm, true! It does look like stack building will work correctly when listing all resource stacks. Single resource lookups will still only yield that first entry, though- might be a weird inconsistency (also from previous behaviour).